### ISSUE TITLE
[feature idea] use whisper for resync

### ISSUE THREAD

**Issue by 0wwafa** [SubtitleEdit/subtitleedit#9056]
A great feature would be to use whisper not to do speech to text but to resync subtitles.

Sometimes I have subtitles but they are for a different version of the video.
Sometime frame rate conversion doesn't work with these files.

It would be great to be able to use whisper (perhaps even a small model will do) to get the sync points and resync the currently loaded subs.

What do you think?

**Comment by darnn** [comment]
@0wwafa This kinda does what you're interested in:
https://github.com/EtienneAb3d/WhisperTimeSync
It's not perfect, you do have to fix some things manually after the fact (words ending up in the wrong subtitle), but it beats doing it manually. All you need from that page is the WhisperTimeSync.jar file. Two things to bear in mind: It chokes on long files, so try to split whatever you're feeding it into 900-subtitle chunks, and, once you get its output SRT, open it to make sure the first word of the first subtitle hasn't ended up before the number and timecode, which sometimes happens.

**Comment by 0wwafa** [comment]
@darnn glad there is a different program.. I just suggested to add the feature to SubTitleEdit.

**Comment by darnn** [comment]
Well, sure, but given that Nik closed the issue with the wontfix tag, I think that's kind of a non-starter. And WhisperTimeSync's really easy to use, you just have a command like this:
`"C:\WhisperTimeSync\distrib\WhisperTimeSync.jar" "D:\Dropbox\Words of Remembrance full V2_en.srt" "D:\Dropbox\remembrance.txt" en`
And plug in the appropriate files whenever you want to do it. I've done it many times and didn't feel put out by not having it be in SE directly.

**Comment by 0wwafa** [comment]
> Well, sure, but given that Nik closed the issue with the wontfix tag, I think that's kind of a non-starter. And WhisperTimeSync's really easy to use, you just have a command like this: `"C:\WhisperTimeSync\distrib\WhisperTimeSync.jar" "D:\Dropbox\Words of Remembrance full V2_en.srt" "D:\Dropbox\remembrance.txt" en` And plug in the appropriate files whenever you want to do it. I've done it many times and didn't feel put out by not having it be in SE directly.

no.. I mean to resync subtitles using a video!

`./command Words of Remembrance full V2_en_wrong_timings.srt Words of Remembrance.mp4
`

**Comment by darnn** [comment]
Yes. What I'm talking about is just two steps instead of one. First you obtain the (more or less) correct timing using Whisper. Then you use WhisperTimeSync as above.
